libraries/botframework-schema/src/teams/index.ts

```diff
@@ -1225,6 +1229,175 @@ export interface FileConsentCardResponse {
+export interface TabResponse {
+    /**
+     * @member {TabResponsePayload} [tab] The response to the tab/fetch message.
+     */
+    tab: TabResponsePayload;
+}
+
+/**
+ * @interface
+ * Envelope for cards for a Tab request.
+ *
+ */
+export interface TabResponseCard {
+    /**
+     * @member {Record<string, any>} [card] The adaptive card for this card tab response.
+     */
+    card: Record<string, unknown>;
+}
+
+/**
+ * @interface
+ * Envelope for cards for a TabResponse.
+ *
+ */
+export interface TabResponseCards {
+    /**
+     * @member {TabResponseCard[]} [cards] Adaptive cards for this card tab response.
+     */
+    cards: TabResponseCard[];
+}
+
+/**
+ * @interface
+ * Payload for Tab Response.
+ *
+ */
+export interface TabResponsePayload {
+    /**
+     * @member {'continue' | 'auth' | 'silentAuth'} [type] Choice of action options when responding to the tab/fetch message.
+     */
+    type?: 'continue' | 'auth' | 'silentAuth';
+    /**
+     * @member {TabResponseCards} [value] The TabResponseCards to send when responding to
+     * tab/fetch activity with type of 'continue'.
+     */
+    value?: TabResponseCards;
+    /**
+     * @member {TabSuggestedActions} [suggestedActions] The Suggested Actions for this card tab.
+     */
+    suggestedActions?: TabSuggestedActions;
+}
+
+/**
+ * @interface
+ * Invoke ('tab/submit') request value payload.
+ *
+ */
+export interface TabSubmit {
+    /**
+     * @member {TabEntityContext} [tabContext] The current tab's entity request context.
+     */
+    tabContext?: TabEntityContext;
+    /**
+     * @member {TabContext} [context] The current user context, i.e., the current theme.
+     */
+    context?: TabContext;
+    /**
+     * @member {TabSubmitData} [data] User input. Free payload containing properties of key-value pairs.
+     */
+    data?: TabSubmitData;
+}
+
+/**
+ * @interface
+ * Invoke ('tab/submit') request value payload data.
+ *
+ */
+export interface TabSubmitData {
+    /**
+     * @member {string} [type] Should currently be 'tab/submit'.
+     */
+    type?: string;
+    /**
+     * @member {any} [any] Additional properties not otherwise defined in TabSubmit.
+     */
+    [properties: string]: unknown;
+}
+
+/**
+ * @interface
+ * Tab SuggestedActions (Only when type is 'auth' or 'silentAuth').
+ *
+ */
+export interface TabSuggestedActions {
+    /**
+     * @member {CardAction[]} [actions] Actions to show in the card response.
+     */
+    actions: CardAction[];
+}
+
+/**
+ * @interface
+ * Tab response to 'task/submit'.
+ *
+ * @extends TaskModuleResponseBase
+ */
+export interface TaskModuleCardResponse extends TaskModuleResponseBase {
+    /**
+     * @member {string} [value] JSON for Adaptive cards to appear in the tab.
+     */
+    value?: string;
+}
+
+/**
+ * A string, representing the task module's height in pixels, one of: small, medium, large.
+ *
+ * @readonly
+ * @enum {string}
+ */
+export type TeamsTaskModuleSize = 'small' | 'medium' | 'large';
```

Comment:
@mdrichardson thanks for adding this! However, since this type is only used in two members of a specific interface, let's get rid of the type and just inline it to make it easier for developers and align with what you did for [TabResponsePayload.type](https://github.com/microsoft/botbuilder-js/pull/3224/files#diff-348dcf11b9d343d90e766394f8ebae3c3675bde51e95705520bd15451f0c5720R1320)

So in TaskModuleTaskInfo, `height` and `width` should look like the following:

```ts

  /**
   * @member {any} [height] This can be a number, representing the task
   * module's height in pixels, or a string, one of: small, medium, large.
   */
  height?: number | 'small' | 'medium' | 'large';

  /**
   * @member {any} [width] This can be a number, representing the task module's
   * width in pixels, or a string, one of: small, medium, large.
   */
  width?: number | 'small' | 'medium' | 'large';
```

Speaking from personal experience, IntelliSense doesn't auto-expand this for developers.